src/Compilers/CSharp/Portable/Lowering/LocalRewriter/PipelinePhaseValidator.cs

```diff
@@ -0,0 +1,107 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.CSharp;
+
+#if DEBUG
+internal enum PipelinePhase
+{
+    InitialBinding,
+    LocalRewriting,
+    Spilling,
+    ClosureConversion,
+    StateMachineRewriting,
+    Emit
+}
+#endif
+
+internal sealed partial class PipelinePhaseValidator
+#if DEBUG
+     : BoundTreeWalkerWithStackGuardWithoutRecursionOnTheLeftOfBinaryOperator
+#endif
+{
+
+#if DEBUG
+    private readonly PipelinePhase _completedPhase;
+#endif
+
+#if DEBUG
+    private PipelinePhaseValidator(PipelinePhase completedPhase)
```

Comment:
Should we have a private constructor outside DEBUG too, perhaps a throwing one? (I think you've added it previously, but it seems removed now.)